extractor/src/main/java/org/schabi/newpipe/extractor/utils/Base64.java

```diff
@@ -0,0 +1,725 @@
+package org.schabi.newpipe.extractor.utils;
+
+import java.io.UnsupportedEncodingException;
+
+/**
+ * For Java 6-7, one could use the {@link javax.xml.bind.DatatypeConverter} to
+ * work with Base64 encoding/decoding. However the javax.xml.bind.* package is
+ * omitted on Android, so we cannot use this class on Android.
+ *
+ * Hence below implementation was picked from AOSP here ->
+ * https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/android/util/Base64.java
```

Comment:
Depends on the license. In case of the Apache 2.0 license, I think you must:

* copy the original license header (contents) into this source file
* include a copy of the license somewhere in the repository
* highlight changes in the code "prominently" (i.e., in a comment)
* in case AOSP contains a `NOTICE` file, there's some more rules

The Apache 2.0 license is quite complex to get right. You should think about using a different implementation.

Reply:
I am not well versed with licenses so using this seemed like a no-brainer. Their is no straight-forward way of performing Base64 encoding when restricted to Java 7. There is the Apache commons library which provides this functionality but adding a library dependency for this singular use case seemed excessive.

Reply:
@Bluesir9 This is definetely a good intention and keeps the extractor small.
@TheAssassin I just went through the [NOTICE file](https://github.com/aosp-mirror/platform_frameworks_base/blob/master/NOTICE) and couldn't find any specific paragraph which matches the base64.java file, because it was written for/by the AOSP. Additionally, @Bluesir9  didn't change the code. So I guess, only these two steps are required:

> - copy the original license header (contents) into this source file
> - include a copy of the license somewhere in the repository

Reply:
@TobiGr If this is all then I will go ahead and make the change in sometime.

Reply:
@TobiGr Uh I have added the original license header in the Base64 class file as per your first point. What license needs to be copied into the repo for the second point?

Reply:
The license starts [here](https://github.com/aosp-mirror/platform_frameworks_base/blob/47d385a5b3c76111dedef11cb814486dacd77b8c/NOTICE#L108). I'd suggest to create a new 3rd-party-licenses directory which contains all the licenses apart from GPL3. @TheAssassin What do you think?

Reply:
I guess that should work.